cmd/fs-v1.go

```diff
@@ -592,6 +593,9 @@ func (fs *FSObjects) getObject(ctx context.Context, bucket, object string, offse
 	buf := make([]byte, int(bufSize))
 
 	_, err = io.CopyBuffer(writer, io.LimitReader(reader, length), buf)
+	if err == io.ErrClosedPipe {
```

Comment:
io.ErrClosedPipe is an error we shouldn't ignore it, why do you think its a good condition? This indicates a bug.

Reply:
incase of range queries, the writer passed to the `getObject` is closed when the range/length is satisfied. This emits `io.ErrClosedPipe` which indicates that the data for the given range is completely read.
The writer is closed here in the handler
https://github.com/Praveenrajmani/minio/blob/9f54f0ad903b51e4b4e6fc3a2b95785fa41442e0/cmd/object-handlers.go#L221

Reply:
Since the `compressed-hrange` is not known, range is set in the handler while decompressing. ( on the decompressed size / actual size )

Reply:
Else we can ignore it only for compressed objects , by checking with `isCompressed()` ?? . But again this introduces a change in object layer.